demes/load_dump.py

```diff
@@ -19,10 +19,12 @@ def _open_file_polymorph(polymorph, mode="r"):
     just yield polymorph under the assumption it's a fileobj.
     """
     try:
-        with open(polymorph, mode) as f:
-            yield f
+        f = open(polymorph, mode)
     except TypeError:
-        yield polymorph
+        f = polymorph
+    yield f
```

Comment:
the file won't be closed if an exception occurs during yield, how about 
```python
try:
     yield f
finally:
    if f is not polymorph:
        f.close()
```

Reply:
Huh. That's confusing, but it seems you're right!